joomla/joomla-cms
# [4.0] Fix Custom Classes for Radio Fields

Pull Request for https://github.com/joomla/joomla-cms/pull/33176#issuecomment-831180478
Thanks for bringing it into my attention @regularlabs 

### Summary of Changes
Reverted `$divClass` back to `$class` for styling the outer div


### Testing Instructions
Make a new plugin XML form with a `radio` field or use an existing one for example:  `/plugins/content/vote/vote.xml`
https://github.com/joomla/joomla-cms/blob/8a1d803f59a669dfe4574867bf5e5b7d751f6d91/plugins/content/vote/vote.xml#L33

Replace the Radio field with the following code:
```
<field
	name="show_total_votes"
	type="radio"
	label="PLG_VOTE_TOTAL_VOTES_LABEL"
	class=""
	default="0"
	filter="integer"
>
	<option value="0">JHIDE</option>
	<option value="1">JSHOW</option>
</field>
```
Try out the following combinations for the `class=""` attribute: (Feel free to replace text-muted with any other custom class)

1. "btn-group"
2. "btn-group text-muted"
3. "btn-group btn-group-yesno"
4. "btn-group btn-group-yesno text-muted"
5. "text-muted"
6. And lastly, remove the class attribute and check

### Actual result BEFORE applying this Pull Request
The Custom Class (text-muted) will not affect the output and it solely depends on `btn-group` and `btn-group-yesno`


### Expected result AFTER applying this Pull Request
The custom class will also be taken into consideration 

https://user-images.githubusercontent.com/53610833/116872439-81726700-ac33-11eb-8b58-78272893bf38.mp4


### Documentation Changes Required

1. In [Joomla 4 Docs Plugin Tutorial Page](https://docs.joomla.org/J4.x:Creating_a_Plugin_for_Joomla), the XML form has mentioned `class="switcher"` for the radio field. This class is redundant and doesn't affect anything. I believe the author intended to write `layout="joomla.form.field.radio.switcher"`

2. While studying the code, I found out that classes, (Edit: also onchange and onclick) can also be applied to the `<option>` as `<option class="text-muted">` under the radio field. For example:
```
<field
	name="show_total_votes"
	type="radio"
	label="PLG_VOTE_TOTAL_VOTES_LABEL"
	default="0"
	filter="integer"
>
	<option value="0" class="text-muted">JHIDE</option>
	<option value="1">JSHOW</option>
</field>
```
will dislpay the output as:

![image](https://user-images.githubusercontent.com/53610833/116873122-c1861980-ac34-11eb-8c69-831291492620.png)

The documentations doesn't mention this: (only attribute given here is value) so a new attribute for `class` (Edit: also `onchange` and `onclick`) must be mentioned here

![image](https://user-images.githubusercontent.com/53610833/116873248-fbefb680-ac34-11eb-8079-7b5fac97042a.png)

Link: https://docs.joomla.org/Radio_form_field_type

The code that is responsible for checking the option tag's classes is: 
https://github.com/joomla/joomla-cms/blob/8a1d803f59a669dfe4574867bf5e5b7d751f6d91/layouts/joomla/form/field/radio/buttons.php#L113
